twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -36,452 +39,570 @@
-
-        $tiqrResponse = $this->session->get("pi_TIQR_Response");
-        $pushResponse = $this->session->get("pi_PUSH_Response");
-        if ($pushResponse || $tiqrResponse) {
-            if ($tiqrResponse) {
-                $template->assign("tiqrImage", $this->session->get("pi_TIQR_Image"));
-            }
-            $template->assign("response", true);
-            $template->assign("responseStatus", $this->session->get("pi_Response_Status"));
+        if ($this->session->get("pi_loadCounter") !== null)
+        {
+            $loads = $this->session->get("pi_loadCounter");
         }
-        return $template;
-    }
+        $template->assign("loadCounter", $loads);
 
-    /**
-     * Return an associative array that contains the options that should be passed to
-     * the HTTP client service when creating HTTP requests.
-     * @return array
-     */
-    private function getClientOptions()
-    {
-        $checkssl = $this->getAppValue('checkssl', '');
-        $noproxy = $this->getAppValue('noproxy', '');
-        $timeout = $this->getAppValue('pitimeout', '5');
-        $options = ['headers' => ['user-agent' => "ownCloud Plugin"],
-            // NOTE: Here, we check for `!== '0'` instead of `=== '1'` in order to verify certificates
-            // by default just after app installation.
-            'verify' => $checkssl !== '0',
-            'debug' => false,
-            'exceptions' => false,
-            'timeout' => $timeout];
-        if ($noproxy === "1") {
-            $options["proxy"] = ["https" => "", "http" => ""];
-        }
-        return $options;
+        return $template;
     }
 
     /**
      * Verify the given challenge.
-     * In fact it is not a challenge but the OTP value!
      *
      * @param IUser $user
      * @param string $challenge
-     * @return Boolean, True in case of success
-     */
-    public function verifyChallenge(IUser $user, $challenge)
-    {
-        $this->log("debug", "privacyIDEA challenge= " . $challenge);
-        return $this->authenticate($user->getUID(), $challenge);
-    }
-
-    /**
-     * Try to authenticate the given username with the given password.
-     *
-     * @param string $username
-     * @param string $password
-     * @return Boolean, True in case of success. In case of failure, this raises
+     * @return Bool True in case of success. In case of failure, this raises
      *         a TwoFactorException (for OC >= 9.2) with a descriptive error message.
-     *         If privacyIDEA returned a HTTP 200 and result->status=true, but
+     *         If privacyIDEA returned an HTTP 200 and result->status=true, but
      *         result->value=false, the exception has a code 1 (i.e. if the user
      *         could be found, but the password was incorrect).
      * @throws TwoFactorException
      */
-    public function authenticate($username, $password)
+    public function verifyChallenge(IUser $user, $challenge): bool
```

Comment:
add type definition for $challenge

Reply:
I can't because of the declaration in the core. Here is the error: Declaration must be compatible with IProvider->verifyChallenge(user: \OCP\IUser, challenge: string). It is especially strange because in other places it is possible. Only here not if I good remember.